### DISCUSSION TITLE
do I need to use `json()` for just returning the data

### DISCUSSION THREAD

**Discussion by lili21** [remix-run/remix#3492] [Q&A] [answered]
```js
export const loader = () => {
  return { hello: 'world' }
}
```

it works as expected. But I wanna know the different with `json`

```js
export const loader = () => {
  return json({ hello: 'world' })
}
```

**Comment by sergiodxa** [comment, marked as answer]
The json functions returns a Response object, and aside of data you can pass a ResponseInit as second argument to customize the status code and headers.

When you don’t return a Response from the loader or action, Remix will pass your value to the json function and create the response for you.

**Reply by lili21** [reply]
is it a best practice to always wrap the data with `json` function even there is no second argument?

**Reply by kiliman** [reply]
Yes, I think so. It's definitely more explicit that you're returning a JSON response.
